### PR TITLE
Decouple group tests

### PR DESCRIPTION
**Pull request by grebneerg**
## Overview
Make the group tests able to be run independently of each other.

## Changes Made
Tests rewritten to not depend on global state and to all clean up after themselves.

## Test Coverage
All tests pass consistently and when run individually.

## Next Steps
This still needs to be done for the tests of most other parts of the codebase.

## Related PRs or Issues (delete if not applicable)
Part of #251

### CODE REVIEW

File: test/repos/group.test.js [hunk 1]
```diff
@@ -21,33 +24,41 @@ beforeAll(async () => {
...
   expect(group2.admins.length).toEqual(0);
   expect(group.members.length).toEqual(0);
-  uuid2 = group2.uuid;
 });
 
 test('Get Group by ID', async () => {
-  const group = await GroupsRepo.getGroupByID(uuid);
-  expect(group.name).toBe('Group');
-  expect(group.code).toBe(code);
+  const g = await GroupsRepo.getGroupByID(uuid);
```
Comment (aliciaxw) [inline test/repos/group.test.js]: should be consistent across all tests, we can use `group` or `g` but maybe not both

File: test/repos/group.test.js [hunk 2]
```diff
@@ -145,23 +154,29 @@ test('Get Members of Group', async () => {
...
   expect(users2.length).toEqual(2);
 });
 
 test('Remove Member from Group', async () => {
-  const group = await GroupsRepo.removeUserByGroupID(uuid, [user2.uuid], 'member');
-  expect(group.members.length).toEqual(0);
-  ({ uuid } = group);
+  const g = await GroupsRepo.removeUserByGroupID(uuid, [user2.uuid], 'member');
+  expect(g.members.length).toEqual(0);
+  // ({ uuid } = group);
```
Comment (aliciaxw) [inline test/repos/group.test.js]: delete

File: test/routes/api.group.test.js [hunk 3]
```diff
@@ -78,9 +85,9 @@ test('Get admins for group', async () => {
   await request(get(`/sessions/${group.id}/admins/`, adminToken)).then((getres) => {
     expect(getres.success).toBe(true);
     const admins = getres.data;
-    expect(admins.length).toBe(2);
+    console.log('hmm');
```
Comment (aliciaxw) [inline test/routes/api.group.test.js]: delete

File: test/repos/group.test.js [hunk 4]
```diff
@@ -21,33 +24,41 @@ beforeAll(async () => {
...
   user = await UsersRepo.createDummyUser('grouptest1');
+  user2 = await UsersRepo.createDummyUser('grouptest2');
+  user3 = await UsersRepo.createDummyUser('grouptest3');
+  user4 = await UsersRepo.createDummyUser('grouptest4');
+
+  group = await GroupsRepo.createGroup('Group', code, user);
+  group2 = await GroupsRepo.createGroup('NewGroup', code2);
+
+  ({ uuid } = group);
+  uuid2 = group2.uuid;
```
Comment (yuna878) [inline test/repos/group.test.js]: Making it consistent will probably make it more readable

File: test/repos/group.test.js [hunk 5]
```diff
@@ -71,47 +82,43 @@ test('Get Admins from Group', async () => {
...
   const { admins } = (await GroupsRepo.addUsersByGoogleIDs(uuid, [user2.googleID], 'admin'));
   expect(admins.length).toEqual(2);
   expect(admins[1].uuid).toBe(user2.uuid);
 
-  const group = await GroupsRepo.removeUserByGroupID(uuid, [user2.uuid], 'admin');
-  expect(group.admins.length).toEqual(1);
-  ({ uuid } = group);
+  const g = await GroupsRepo.removeUserByGroupID(uuid, [user2.uuid], 'admin');
+  expect(g.admins.length).toEqual(1);
+  // ({ uuid } = group);
```
Comment (yuna878) [inline test/repos/group.test.js]: Delete line if unused

File: test/repos/group.test.js (hunk 1 above)
Comment (grebneerg) [reply test/repos/group.test.js]: In these tests, `group` refers to a global group object managed by `beforeEach` and `afterEach`, while g refers to instances created or retrieved within the test itself.

**Pull request merged**